app/src/main/java/com/jnj/vaccinetracker/reportsoverview/vaccinesoverview/screens/VaccinesOverviewFragment.kt

```diff
@@ -303,35 +326,93 @@ class VaccinesOverviewFragment : BaseFragment(),
         }
     }
 
+
+    @SuppressLint("SetTextI18n")
     private fun applyFilters(
-        vaccinesData: List<VaccineObservationDTO> = vaccinesOverviewViewModel.vaccineDTOs.value
-            ?: emptyList()
+        vaccinesData: List<VaccineObservationDTO> = vaccinesOverviewViewModel.vaccineDTOs.value ?: emptyList()
     ) {
+        val isStatic = { loc: String? -> loc.equals("static", ignoreCase = true) }
+        val isOutreach = { loc: String? ->
+            loc?.let {
+                it.equals("outreach", ignoreCase = true) ||
+                        it.equals("outreach immunization", ignoreCase = true) ||
+                        it.equals("outreach session", ignoreCase = true)
+            } ?: false
+        }
+        val isSchool = { loc: String? -> loc.equals("school", ignoreCase = true) }
+
         val filteredData = vaccinesData.filter { observation ->
+
             val dateMatches = if (observation.administerDate.isNotEmpty()) {
-                (selectedStartDate == null || DateUtil.convertStringToDate(
+                val observationDate = DateUtil.convertStringToDate(
                     observation.administerDate,
                     DateFormat.FORMAT_DATE.toString()
-                )!! >= selectedStartDate?.toDate()) &&
-                        (selectedEndDate == null || DateUtil.convertStringToDate(
-                            observation.administerDate,
-                            DateFormat.FORMAT_DATE.toString()
-                        )!! <= selectedEndDate?.toDate())
-            } else {
-                false
+                ) ?: return@filter false
```

Comment:
what does it mean? return@filter false

Reply:
starting from above statement , the if checks if **observation.administerDate** is not empty.
It then tries to convert **observation.administerDate** to a Date object using **DateUtil.convertStringToDate.**
If the conversion fails (returns null), **return@filter false** is executed, which excludes the current observation from the filtered list.
If the conversion succeeds, it checks if the **observationDate** is within the selected date range.
The result of these checks is assigned to dateMatches.